src/diffusers/models/autoencoders/autoencoder_DCAE.py

```diff
@@ -0,0 +1,542 @@
+# Copyright 2024 MIT, Tsinghua University, NVIDIA CORPORATION and The HuggingFace Team.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import Any, Optional, Callable
+
+import torch
+import torch.nn as nn
+from omegaconf import MISSING, OmegaConf
+from huggingface_hub import PyTorchModelHubMixin
+
+from .dc_ae_blocks.act import build_act
+from .dc_ae_blocks.norm import build_norm
+from .dc_ae_blocks.ops import (
+    ChannelDuplicatingPixelUnshuffleUpSampleLayer,
+    ConvLayer,
+    ConvPixelShuffleUpSampleLayer,
+    ConvPixelUnshuffleDownSampleLayer,
+    EfficientViTBlock,
+    IdentityLayer,
+    InterpolateConvUpSampleLayer,
+    OpSequential,
+    PixelUnshuffleChannelAveragingDownSampleLayer,
+    ResBlock,
+    ResidualBlock,
+)
+
+__all__ = ["DCAE", "dc_ae_f32c32", "dc_ae_f64c128", "dc_ae_f128c512"]
+
+
+@dataclass
+class EncoderConfig:
+    in_channels: int = MISSING
+    latent_channels: int = MISSING
+    width_list: tuple[int, ...] = (128, 256, 512, 512, 1024, 1024)
+    depth_list: tuple[int, ...] = (2, 2, 2, 2, 2, 2)
+    block_type: Any = "ResBlock"
+    norm: str = "rms2d"
+    act: str = "silu"
+    downsample_block_type: str = "ConvPixelUnshuffle"
+    downsample_match_channel: bool = True
+    downsample_shortcut: Optional[str] = "averaging"
+    out_norm: Optional[str] = None
+    out_act: Optional[str] = None
+    out_shortcut: Optional[str] = "averaging"
+    double_latent: bool = False
+
+
+@dataclass
```

Comment:
We don't rely on `dataclass` for config. Please follow:
https://github.com/huggingface/diffusers/blob/9366c8f84bfe47099ff047272661786ebb54721d/src/diffusers/models/autoencoders/autoencoder_kl.py#L75

Reply:
Sure.